### PR TITLE
lint in examples folder

### PR DESCRIPTION
**Pull request by CJ-Wright**
Still needs tests

### CODE REVIEW

**Comment by scopatz** [comment]
:+1:

**Comment by jakirkham** [comment]
So I'd actually push to make this part of webservices instead for two reasons. First it already special cases `staged-recipes`. Second the `conda-smithy` linter is designed to also be run locally where maybe this might not make sense; whereas, webservices is only designed to lint PRs at conda-forge where tweaks like this are typical. Does that sound reasonable? Happy to give a pointer as to where to look. 

Other than that, this is in principle a good idea. Just trying to find the right home for it. :)

**Comment by isuruf** [comment]
How about having this in `run_conda_forge_lints` function which is conda-forge specific

**Comment by jakirkham** [comment]
Forgot about that function. Yeah that could be a good option as well.

**Comment by CJ-Wright** [comment]
This now runs in conda-forge lints

**Comment by CJ-Wright** [comment]
@conda-forge/core ready for review/merge

**Comment by mwcraig** [comment]
LGTM

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -265,6 +265,10 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
         except:
             lints.append("Package version {} doesn't match conda spec".format(ver))
 
+    # 18: if the recipe dir is inside the example dir
+    if recipe_dir is not None and 'recipes/example/' in recipe_dir:
+        lints.append('Please move the recipe out of the example dir and '
+                     'into its own dir.')
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Please remove duplicate here.

Comment (CJ-Wright) [reply conda_smithy/lint_recipe.py]: Thank you! (Done)

**Pull request merged**